### ISSUE TITLE
Wrong Environment on multinode jobs

### ISSUE THREAD

**Issue by feine-moehre** [Linaro/squad#553]
SQUAD 0.65, LAVA 2019.04

I ran a multinode job on two DUTs with theses job ids:
![grafik](https://user-images.githubusercontent.com/50700938/59438929-a74eb280-8df4-11e9-8dad-a4fef3d91da7.png)

SQUAD displays as Environment the same DUT twice for the differend job ids which is obviously wrong.
![grafik](https://user-images.githubusercontent.com/50700938/59439056-db29d800-8df4-11e9-8c5e-bc5fcd58ae85.png)

This bug is new in Version 0.65 (see also #533)

**Comment by mwasilew** [comment]
Discussion is recorded in the PR https://github.com/Linaro/squad/pull/535

**Comment by feine-moehre** [comment]
One workaround could be using a delimiter "_" between device names as environment. This will work with version 0.65. 

Maybe you can try to match the splitted parts of the environment to the device types within the lava result. This is also not a satisfying solution because the delimiter restrict the character set of lava device type names.

**Comment by mwasilew** [comment]
I would have to ask LAVA each time which device the job run on. This adds complexity and 'more magic' to the LAVA ci backend. It's also not consistent with how the environments are done right now. 

Let's keep this ticket open and I'll think on some solution. Maybe not all hope is lost :)

**Comment by chaws** [comment]
@mwasilew do you think that the LAVA REST api has something to help fix the environment issue?

**Comment by mwasilew** [comment]
@chaws the problem shows up on fetching results from LAVA. Since the submission is using single environment all resulting test jobs (multinode in this case) use the same env. We would need to either analyze the test job yaml before submitting to LAVA or query LAVA about the actual device-type used. Both solutions suffer from the some problems, for example lava device type naming is not standardized. There can also be aliases which complicate things even more. I really don't have a good solution :(

**Comment by chaws** [comment]
Maybe this helps: https://git.lavasoftware.org/lava/lava/-/merge_requests/1190

**Comment by mwasilew** [comment]
@chaws nope, this is just fix to the lava scheduler. Nothing to do with how SQUAD names environments
